### PR TITLE
feat: add requestWhatsappIntegration api and whatsapp number to dealer profile

### PR DESCRIPTION
**Pull request by ufkaya**
- [CAR-5290 - Cockpit: Whatsapp CTA phone no. entry with confirmation + BO notification](https://autoricardo.atlassian.net/browse/CAR-5290)

### CODE REVIEW

File: src/services/dealer.ts [hunk 1]
```diff
@@ -224,3 +224,31 @@ export const requestMatelsoIntegration = async ({
...
+        ...options,
+      },
+    })
+
+    return {
+      tag: "success",
+      result,
+    }
+  } catch (error) {
+    return handleValidationError(error, { swallowErrors: true })
```
Comment (masone) [inline src/services/dealer.ts:252]: can you help me understand - what errors do we swallow and why?
what validation errors could occur in this case?

**Review by masone — COMMENTED** [review]
some questions on the error handling :)

File: src/services/dealer.ts (hunk 1 above)
Comment (Averethel) [reply src/services/dealer.ts:252]: If you add `{ swallowErrors: true }` response errors other than 400 and 422 (as well as other exceptions) will be returned as:
```
{
  tag: "error",
  message: "validation.other-error",
  errors: [],
}
```

that's how we typically handle non-critical API requests
(check the code for reference: https://github.com/carforyou/carforyou-api-client-pkg/blob/master/src/base.ts#L192)

**Review by Averethel — APPROVED** [review]

File: src/services/dealer.ts (hunk 1 above)
Comment (ufkaya) [reply src/services/dealer.ts:252]: 1. I used the same code (copy/paste) and I didnt check what that part does, thanks @Averethel 
2. we send the whatsappNumber, and we get the validation error if not populated

**Review by ralucacril — APPROVED** [review]

File: src/services/dealer.ts (hunk 1 above)
Comment (masone) [reply src/services/dealer.ts:252]: ah, so this is going to be used in a form... thanks for the explanation, makes sense

**Review by masone — APPROVED** [review]

**Pull request merged**